backend/routers/admin_products.py

```
@@ -378,6 +387,101 @@ async def delete_credit_package(package_id: str, db=Depends(get_db)):
         return {"success": False, "error": "Credit package not found"}, 404
     return {"success": True}
 
+# --- PROKERALA SMART PRICING ENDPOINTS ---
+@router.get("/pricing/prokerala-cost/{service_id}")
+async def get_prokerala_cost_analysis(
+    service_id: int,
+    user_id: Optional[str] = None,
+    db=Depends(get_db)
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)